sktime/transformations/series/compose.py

```diff
@@ -96,3 +97,210 @@ def inverse_transform(self, Z, X=None):
+
+     Attributes
+    ----------
+    transformers_ : dict of {str : transformer}
+        Maps columns to transformers
+
+    Example
+    -------
+    >>> from sktime.datasets import base
+    >>> from sktime.transformations.series.detrend import Detrender
+    >>> from sktime.transformations.series.compose import ColumnwiseTransformer
+
+    >>> y, X = base.load_longley()
+    >>> transformer = ColumnwiseTransformer(Detrender())
+    >>> yt = transformer.fit_transform(X)
+    """
+
+    _required_parameters = ["transformer"]
+
+    def __init__(self, transformer, columns=None):
+        self.transformer = transformer
+        self.columns = columns
+
+        # check that columns are None or list of strings
+        if columns is not None:
+            if not isinstance(columns, list) and all(
+                isinstance(s, str) for s in columns
+            ):
+                raise ValueError("Columns need to be a list of strings or None.")
+
+        self.columns_ = columns
+        self.transformers_ = None
+        super(ColumnwiseTransformer, self).__init__()
+
+    def fit(self, Z, X=None):
+        """
+        Iterates over columns (series) and applies
+        the fit function of the transformer.
+
+        Parameters
+        ----------
+        Z : pd.Series, pd.DataFrame
+
+        Returns
+        -------
+        self : an instance of self
+
+        """
+        z = check_series(Z, allow_numpy=False)
+        self._is_fitted = False
+
+        # univariate case
+        if isinstance(z, pd.Series):
+            z = z.to_frame()
+
+        if self.columns_ is None:
+            self.columns_ = z.columns
+
+        # make sure z contains all columns that the user wants to transform
+        self._check_columns(z)
+
+        self.transformers_ = {}
+        for colname in self.columns_:
+            transformer = clone(self.transformer)
+            self.transformers_[colname] = transformer
+            self.transformers_[colname].fit(z[colname], X)
+        self._is_fitted = True
+        return self
+
+    def transform(self, Z, X=None):
+        """Transform data.
+        Returns a transformed version of Z by iterating over specified
+        columns and applying the univariate series transformer to them.
+
+        Parameters
+        ----------
+        Z : pd.Series, pd.DataFrame
+
+        Returns
+        -------
+        Z : pd.Series, pd.DataFrame
+            Transformed time series(es).
+        """
+        self.check_is_fitted()
+        z = check_series(Z)
+        is_series = False
+
+        # univariate case
+        if isinstance(z, pd.Series):
+            is_series = True
+            name_of_z = z.name
+            z = z.to_frame()
+
+        # make copy of z
+        z = z.copy()
+
+        # make sure z contains all columns that the user wants to transform
+        self._check_columns(z)
+        for colname in self.columns_:
+            # self.columns_ : columns that are supposed to be transformed
```

Comment:
I think this docstring can be put to the top where this attribute is initiated. Actually if we keep `self.columns_` we would ideally need also an `Attributes` docstring for it like
```
Attributes
----------
columns_ : something
     My desciption

```